### PR TITLE
Add activity endpoints

### PR DESCRIPTION
**Pull request by sthuray**
## Notion ticket link

[ Pet Activity Endpoints ](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* Implemented CRUD operations for activities table
* Removed create and update columns for activities table by setting "timestamps: false"
* Ensured proper error handling using NotFoundError
* ID parameter is a number except when returning deleted id

## Steps to test
1. Test CRUD operations using Postman:
    a. Creating an activity returns created activity and updates database
Create Activity
    b. Getting all activities returns all activities in database
Get Activities
    c1. Getting an activity by id returns correct activity
Get Activity 2
    c2. Getting an activity with non-existent id returns a 404 error
Get Activity 3
    d1. Updating an activity returns updated activity and updates database
Update Activity 1
    d2. Updating an activity with non-existent id returns a 404 error
Update Activity 10
    e1. Deleting an activity returns deleted id and updates database
Delete Activity 1
    e2. Deleting an activity with a non-existent id returns a 404 error
Delete Activity 30
    e3. Updating an activity after deleting associated id returns a 404 error
Update Activity 1 After Deleting 1
    e4. Getting all activities after deleting an activity returns updated activities
Get Activities After Deleting 1
    e5. Getting all activites after deleting all activites returns empty JSON
Get Activities After Deleting 1 and 2

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by aanxniee — APPROVED** [review]
lgtm @MinIcecream let's have daniel review this

**Review by jerry-cheng5 — APPROVED** [review]
Looks good to me! Tests look thorough as well :)

**Review by MinIcecream — APPROVED** [review]
Yeah, looks good to me!

File: backend/typescript/rest/activityRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const { body } = req;
+    const newActivity = await activityService.createActivity({
+      activityName: body.activityName,
+    });
+    res.status(201).json(newActivity);
+  } catch (e: unknown) {
+    if (e instanceof NotFoundError) {
+      res.status(404).send(getErrorMessage(e));
+    } else {
+      res.status(500).send("Internal server error occured.");
```
Comment (aanxniee) [inline backend/typescript/rest/activityRoutes.ts:27]: let's wait until jerry's pr is in and then change this to use his constant: `INTERNAL_SERVER_ERROR_MESSAGE`

**Review by laks0407 — APPROVED** [review]
looks great - love the detailed tests 👍

**Pull request merged**